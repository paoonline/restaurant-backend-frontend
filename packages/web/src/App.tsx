import { useEffect } from "react";
import { restaurantService } from "./service/restaurantService";
import Shop from "./module/shop";
import "./root.css";

import { createTheme, ThemeProvider } from "@mui/material";
import useMenu from "./hooks/useMenu";

const theme = createTheme({
  typography: {
    fontFamily: '"Roboto", "Arial", sans-serif', // Set your desired font family
  },
});

function App() {
  // useEffect(() => {
  //   restaurantService(567051);
  // }, []);

  const {shopId, onSetShopId} = useMenu()

  return (
    <ThemeProvider theme={theme}>
      {!shopId  && <Shop callback={onSetShopId}/>}
    </ThemeProvider>
  );
}

export default App;
