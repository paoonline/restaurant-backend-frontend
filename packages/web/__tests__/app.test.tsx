import React from "react"
import { render } from "@testing-library/react";
import App from '../src/App'

describe('App', () => {
    it('should render correctly', () => {
        const component = render(<App/>)
        expect(component).toMatchSnapshot()
    })
})