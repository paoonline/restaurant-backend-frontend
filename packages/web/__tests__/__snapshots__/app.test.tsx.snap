// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-15z375g"
      >
        <div
          class="MuiBox-root css-18jocpe"
        >
          <div
            class="MuiBox-root css-1uzu2t1"
          >
            ร้านลืมเคี้ยว
          </div>
        </div>
        <div
          class="MuiBox-root css-18jocpe"
        >
          <div
            class="MuiBox-root css-1uzu2t1"
          >
            Ekkamai Macchiato - Home Brewer
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-15z375g"
    >
      <div
        class="MuiBox-root css-18jocpe"
      >
        <div
          class="MuiBox-root css-1uzu2t1"
        >
          ร้านลืมเคี้ยว
        </div>
      </div>
      <div
        class="MuiBox-root css-18jocpe"
      >
        <div
          class="MuiBox-root css-1uzu2t1"
        >
          Ekkamai Macchiato - Home Brewer
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
